be/src/exprs/agg/distinct.h

```diff
@@ -157,13 +163,18 @@ struct DistinctAggregateStateV2<PT, FixedLengthPTGuard<PT>> {
 
     int64_t disctint_count() const { return set.size(); }
 
-    size_t serialize_size() const { return set.size() * sizeof(T) + sizeof(size_t); }
+    size_t serialize_size() const {
+        size_t size = set.size() * sizeof(T) + sizeof(size_t);
+        size = std::max(size, MIN_SIZE_OF_HASH_SET_SERIALIZED_DATA);
+        return size;
+    }
 
     void serialize(uint8_t* dst) const {
         size_t size = set.size();
         memcpy(dst, &size, sizeof(size));
         dst += sizeof(size);
         for (auto& key : set) {
+            static_assert(sizeof(key) == sizeof(key));
```

Comment:
```suggestion
            static_assert(sizeof(T) == sizeof(key));
```

Reply:
Yes. It's my intention, but we are going to remove it.